packages/fetch-node-details/src/nodeDetailManager.ts

```diff
@@ -100,6 +100,9 @@ class NodeDetailManager {
       const finalKeyType = keyType ?? this._keyType;
       const finalSigType = sigType ?? this._sigType;
 
+      // validate key type and sig type to fetch TSS endpoints
+      validateKeyTypeAndSigTypeForTSS(finalKeyType, finalSigType);
```

Comment:
i think we should do validation on server side only specifically to maintain forward compatibility specially for fnd sdk. Its a good practice to fail fast in general but in this case we want server to be source of truth to ensure older client sdks will remain compatible to future changes in endpoints. I dont know any specific reason to do that for now but its just from past experience we want this sdk to remain future compatible